cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/ProgramStatusEventPublisher.java

```diff
@@ -208,7 +209,14 @@ private void populateErrorDetailsAndMetrics(final ProgramStatusEventDetails.Buil
       return;
     }
     if (status.isEndState()) {
-      CompletableFuture.supplyAsync(() -> metricsProvider.retrieveMetrics(runId))
+      CompletableFuture.supplyAsync(() -> {
+        try {
+          return metricsProvider.retrieveMetrics(runId);
+        } catch (Exception e) {
+          LOG.error("Retries failed for extracting metrics. ", e);
```

Comment:
nit: There are no retries here. Change the message to `Error retrieving metrics from provider`

Reply:
done